Package.swift

```diff
@@ -270,6 +274,11 @@ let package = Package(
         .headerSearchPath("../../.."),
       ]
     ),
+    .target(
+      name: "FirebaseCombineSwift",
+      dependencies: ["FirebaseAuth", "FirebaseFirestore", "FirebaseFirestoreSwift"],
```

Comment:
Hmm, is there a way we can exclude these dependencies so users only get the APIs they need? Or is that a limitation with SwiftPM for now?

Reply:
Maybe you can try this, but I'm not sure it will work:
```suggestion
      dependencies: [
        #if canImport(FirebaseAuth) "FirebaseAuth" #endif, 
        #if canImport(FirebaseFirestore) "FirebaseFirestore" #endif, 
        #if canImport(FirebaseFirestoreSwift) "FirebaseFirestoreSwift" #endif, 
      ],
```

Reply:
Using conditionals like that doesn't work in this context, unfortunately.

There is a proposal out for conditional dependencies with a partial implementation: [SE-273](https://github.com/apple/swift-evolution/blob/main/proposals/0273-swiftpm-conditional-target-dependencies.md), but it doesn't cover our use case at the moment.

I guess this means we'll have to provide individual modules for the respective SDKs.

Reply:
I was thinking of excluding them as dependencies and have "light up" functionality, where the APIs only get surfaced as devs explicitly add each of them. Then we can have a test target that includes all of them. Does that make sense?

Reply:
I think @ryanwilson's solution is the optimal one.

Reply:
Tried out a couple of options, and defining a module for each supported SDK seems to be the best way forward. This has been implemented in https://github.com/firebase/firebase-ios-sdk/pull/6976/commits/2e57702d7fd679acffb3c54f65baa0553111d8cc